build-on-push/jjb/platform-component-setup.yaml

```diff
@@ -0,0 +1,92 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- defaults:
+    # Parameters:
+    #
+    #   * platform-ci-branch: A branch in 'platform-ci' repository used in generated jobs
+    #   * platform-ci-label: Human readable description on the branch in 'platform-ci' repository
+    #   * github-user: A GitHub user whose fork of 'platform-ci' will be used by the generated jobs
+ 
+    name: 'platform-component-setup'
+    display-name: 'Enable CI for a Platform component ({platform-ci-label})'
+    description: |
+        This job can be used to set up the template CI jobs for a Platform
+        component.
+
+        Managed by Jenkins Job Builder. Do not edit via web.
+    concurrent: true
+    node: 'jslave-baseos-build-on-commit'
+    scm:
+      - git:
+          url: 'https://github.com/{github-user}/platform-ci.git'
+          branches:
+            - origin/{platform-ci-branch}
+          basedir: 'platform-ci'
+    parameters:
+      - string:
+          name: COMPONENT
+          default: ''
+          description: 'A component name for which CI settings are altered'
+      - choice:
+          name: BUILD_ON_PUSH
+          choices:
+            - 'do not modify current settings'
+            - 'enable'
+            - 'disable'
+          description: |
+            Set ENABLE if automated builds on git commits pushed to dist-git should be enabled.
+            Set DISABLE if automated builds on git commits pushed to dist-git should be disabled.
+      - string:
+          name: SLAVE
+          default: ''
+          description: 'A name of the slave which will be used to run created jobs'
+    wrappers:
+        - default-ci-workflow-wrappers
+    builders:
+        - shell: |
+            export PATH="$PATH:platform-ci/build-on-push/scripts"
+            export PYTHONPATH="$PYTHONPATH:platform-ci/build-on-push/platform_ci"
+
+            job-platform-component-setup.sh "$COMPONENT" "$BUILD_ON_PUSH" "$SLAVE" "{github-user}" \
+              "{platform-ci-branch}"
+
+- job-template:
+    name: 'platform-component-setup-master'
+    defaults: 'platform-component-setup'
+
+- project:
+    name: platform-component-setup-master
+    project: platform-component-setup-master
+    jobs:
+      - platform-component-setup-master
+    platform-ci-branch: master
+    platform-ci-label: 'Production'
+    github-user: 'RHQE'
+
+# Template for testing instances of the component setup job
+# The job needs to have a different name that other component setup jobs
```

Comment:
I kept them intentionally so it's clear how can one create a "testing" job pulling code from some development location (own fork / branch). I'll fix the typo, thanks!